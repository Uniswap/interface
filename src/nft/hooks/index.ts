export * from './useBag'
export * from './useCollectionFilters'
export * from './useFiltersExpanded'
export * from './useIsCollectionLoading'
export * from './useIsMobile'
export * from './useIsTablet'
export * from './useMarketplaceSelect'
export * from './useNFTList'
export * from './useNFTSelect'
export * from './useProfilePageState'
export * from './useSearchHistory'
export * from './useSelectAsset'
export * from './useSellAsset'
export * from './useSendTransaction'
export * from './useSweep'
export * from './useTransactionResponse'
export * from './useWalletBalance'
export * from './useWalletCollections'
