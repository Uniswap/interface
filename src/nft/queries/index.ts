export * from './genie'
export * from './looksRare'
export * from './openSea'
export * from './x2y2'
