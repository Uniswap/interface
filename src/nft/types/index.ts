export * from './checkout'
export * from './collection'
export * from './common'
export * from './discover'
export * from './navbar'
export * from './sell'
