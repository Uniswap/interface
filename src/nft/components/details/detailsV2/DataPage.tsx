import Column from 'components/Column'
import Row from 'components/Row'
import { GenieAsset } from 'nft/types'
import styled from 'styled-components/macro'
import { BREAKPOINTS } from 'theme'

import { DataPageDescription } from './DataPageDescription'
import { DataPageHeader } from './DataPageHeader'
import { DataPageTable } from './DataPageTable'
import { DataPageTraits } from './DataPageTraits'

const DataPagePaddingContainer = styled.div`
  padding: 24px 64px;
  height: 100vh;
  width: 100%;

  @media screen and (max-width: ${BREAKPOINTS.md}px) {
    height: 100%;
  }

  @media screen and (max-width: ${BREAKPOINTS.sm}px) {
    padding: 24px 48px;
  }

  @media screen and (max-width: ${BREAKPOINTS.xs}px) {
    padding: 24px 20px;
  }
`

const DataPageContainer = styled(Column)`
  height: 100%;
  width: 100%;
  gap: 36px;
  max-width: ${({ theme }) => theme.maxWidth};
  margin: 0 auto;
`

const HeaderContainer = styled.div<{ showDataHeader?: boolean }>`
  position: sticky;
  top: ${({ theme }) => `${theme.navHeight}px`};
  padding-top: 16px;
  backdrop-filter: blur(12px);
  z-index: 1;
  transition: ${({ theme }) => `opacity ${theme.transition.duration.fast}`};
  opacity: ${({ showDataHeader }) => (showDataHeader ? '1' : '0')};

  @media screen and (max-width: ${BREAKPOINTS.md}px) {
    display: none;
  }
`

const ContentContainer = styled(Row)`
  gap: 24px;
  padding-bottom: 45px;

  @media screen and (max-width: ${BREAKPOINTS.lg}px) {
    flex-wrap: wrap;
  }
`

const LeftColumn = styled(Column)`
  gap: 24px;
  width: 100%;
  align-self: flex-start;
`

export const DataPage = ({ asset, showDataHeader }: { asset: GenieAsset; showDataHeader: boolean }) => {
  return (
    <DataPagePaddingContainer>
      <DataPageContainer>
        <HeaderContainer showDataHeader={showDataHeader}>
          <DataPageHeader asset={asset} />
        </HeaderContainer>
        <ContentContainer>
          <LeftColumn>
            {!!asset.traits?.length && <DataPageTraits asset={asset} />}
            <DataPageDescription />
          </LeftColumn>
          <DataPageTable asset={asset} />
        </ContentContainer>
      </DataPageContainer>
    </DataPagePaddingContainer>
  )
}
