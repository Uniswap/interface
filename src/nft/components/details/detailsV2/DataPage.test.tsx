import { TEST_NFT_ASSET } from 'test-utils/nft/fixtures'
import { render } from 'test-utils/render'

import { DataPage } from './DataPage'

it('data page loads with header showing', () => {
  const { asFragment } = render(<DataPage asset={TEST_NFT_ASSET} showDataHeader={true} />)
  expect(asFragment()).toMatchSnapshot()
})

// The header is hidden via opacity: 0 to maintain its spacing, so it still exists in the DOM
// Therefore we can not check for its non-existence and instead rely on comparing the full generated snapshots
it('data page loads without header showing', () => {
  const { asFragment } = render(<DataPage asset={TEST_NFT_ASSET} showDataHeader={false} />)
  expect(asFragment()).toMatchSnapshot()
})
