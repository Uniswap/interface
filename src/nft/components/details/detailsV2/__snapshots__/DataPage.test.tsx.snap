// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`data page loads with header showing 1`] = `
<DocumentFragment>
  .c4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.c12 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  justify-self: flex-end;
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
}

.c14 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: primary;
  border: 0;
  border-radius: 4px;
}

.c5 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c9 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  gap: 4px;
}

.c13 {
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  gap: 12px;
}

.c23 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  gap: 8px;
}

.c10 {
  color: #7780A0;
}

.c11 {
  color: #0D111C;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c20 {
  background: #FFFFFF;
  border: 1px solid #D2D9EE;
  border-radius: 16px;
  padding: 16px 20px;
  width: 100%;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
}

.c21 {
  gap: 32px;
  width: 100;
  margin-bottom: 12px;
  padding-bottom: 12px;
  border-bottom: 1px solid #D2D9EE;
}

.c22 {
  color: #0D111C;
  line-height: 24px;
  cursor: pointer;
}

.c22:hover {
  opacity: 0.6;
}

.c24 {
  color: #98A1C0;
  line-height: 24px;
  cursor: pointer;
}

.c24:hover {
  opacity: 0.6;
}

.c26 {
  background: #D2D9EE;
  border-radius: 4px;
  padding: 2px 4px;
  color: #7780A0;
  line-height: 12px;
}

.c25 {
  height: 252px;
}

.c15 {
  padding: 16px;
  width: 100%;
  font-weight: 500;
  text-align: center;
  border-radius: 20px;
  outline: none;
  border: 1px solid transparent;
  color: #0D111C;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  position: relative;
  z-index: 1;
  will-change: transform;
  -webkit-transition: -webkit-transform 450ms ease;
  -webkit-transition: transform 450ms ease;
  transition: transform 450ms ease;
  -webkit-transform: perspective(1px) translateZ(0);
  -ms-transform: perspective(1px) translateZ(0);
  transform: perspective(1px) translateZ(0);
}

.c15:disabled {
  opacity: 50%;
  cursor: auto;
  pointer-events: none;
}

.c15 > * {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c15 > a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c16 {
  background-color: #F5F6FC;
  color: #7780A0;
  font-size: 16px;
  font-weight: 500;
}

.c16:hover {
  background-color: #d2daf7;
}

.c16:active {
  background-color: #bdc8f3;
}

.c17 {
  white-space: nowrap;
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-radius: 16px;
}

.c6 {
  gap: 24px;
}

.c7 {
  width: 96px;
  height: 96px;
  border-radius: 20px;
  object-fit: cover;
}

.c8 {
  gap: 4px;
  margin-right: auto;
}

.c28 {
  position: relative;
}

.c29 {
  gap: 4px;
  padding: 8px;
  cursor: pointer;
  border-radius: 12px;
  width: 92px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background: #E8ECFB;
  -webkit-transition: opacity 250ms ease;
  transition: opacity 250ms ease;
}

.c29:hover {
  opacity: 0.6;
}

.c29:active {
  opacity: 0.4;
}

.c30 {
  height: 16px;
  width: 16px;
  color: #7780A0;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: -webkit-transform 125ms;
  -webkit-transition: transform 125ms;
  transition: transform 125ms;
}

.c27 {
  height: 568px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c0 {
  padding: 24px 64px;
  height: 100vh;
  width: 100%;
}

.c2 {
  height: 100%;
  width: 100%;
  gap: 36px;
  max-width: 1200px;
  margin: 0 auto;
}

.c3 {
  position: -webkit-sticky;
  position: sticky;
  top: 72px;
  padding-top: 16px;
  -webkit-backdrop-filter: blur(12px);
  backdrop-filter: blur(12px);
  z-index: 1;
  -webkit-transition: opacity 125ms;
  transition: opacity 125ms;
  opacity: 1;
}

.c18 {
  gap: 24px;
  padding-bottom: 45px;
}

.c19 {
  gap: 24px;
  width: 100%;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
}

@media screen and (max-width:1024px) {
  .c7 {
    display: none;
  }
}

@media screen and (max-width:768px) {
  .c0 {
    height: 100%;
  }
}

@media screen and (max-width:640px) {
  .c0 {
    padding: 24px 48px;
  }
}

@media screen and (max-width:396px) {
  .c0 {
    padding: 24px 20px;
  }
}

@media screen and (max-width:768px) {
  .c3 {
    display: none;
  }
}

@media screen and (max-width:1024px) {
  .c18 {
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
  }
}

<div
    class="c0"
  >
    <div
      class="c1 c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4 c5 c6"
        >
          <img
            class="c7"
            src="https://cdn.center.app/1/0xED5AF388653567Af2F388E6224dC7C4b3241C544/3318/50ed67ad647d0aa0cad0b830d136a677efc2fb72a44587bc35f2a5fb334a7fdf.png"
          />
          <div
            class="c1 c8"
          >
            <div
              class="c4 c9"
            >
              <div
                class="c10 css-1aekuku"
              >
                Azuki
              </div>
              <svg
                fill="none"
                height="16px"
                viewBox="0 0 20 20"
                width="16px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4.52795 13.8056C4.52719 14.4043 4.6712 14.8474 4.95997 15.135C5.24798 15.4233 5.68496 15.5651 6.27091 15.5605H7.57497C7.62945 15.5585 7.68379 15.5676 7.73463 15.5873C7.78547 15.607 7.83176 15.6369 7.87062 15.6752L8.79884 16.5928C9.22054 17.0142 9.63382 17.2237 10.0387 17.2214C10.4436 17.2191 10.8569 17.0096 11.2786 16.5928L12.1954 15.6752C12.2356 15.6365 12.2832 15.6063 12.3354 15.5866C12.3876 15.5669 12.4433 15.558 12.499 15.5605H13.7951C14.3871 15.5613 14.8283 15.4171 15.1186 15.1281C15.4089 14.839 15.5541 14.3959 15.5541 13.7987V12.5014C15.5511 12.389 15.5923 12.2799 15.6687 12.1974L16.5854 11.2798C17.0125 10.86 17.2245 10.4467 17.2214 10.0399C17.2184 9.63305 17.0064 9.21935 16.5854 8.79878L15.6687 7.88115C15.592 7.79886 15.5509 7.68965 15.5541 7.57719V6.2799C15.5533 5.68191 15.4093 5.23878 15.1221 4.95049C14.8348 4.66221 14.3925 4.51806 13.7951 4.51806H12.499C12.4433 4.52036 12.3877 4.51138 12.3355 4.49168C12.2834 4.47197 12.2357 4.44193 12.1954 4.40336L11.2786 3.48574C10.8569 3.06439 10.4436 2.85487 10.0387 2.85717C9.63382 2.85946 9.22054 3.06898 8.79884 3.48574L7.87062 4.40336C7.83164 4.44148 7.78536 4.4713 7.73454 4.49101C7.68373 4.51072 7.62943 4.51993 7.57497 4.51806H6.27091C5.67961 4.51883 5.23995 4.66182 4.95194 4.94705C4.66393 5.23228 4.51992 5.67656 4.51992 6.2799V7.58063C4.52314 7.69309 4.48197 7.80229 4.40533 7.88459L3.48859 8.80222C3.06765 9.22203 2.85718 9.63572 2.85718 10.0433C2.85718 10.4509 3.07033 10.8653 3.49662 11.2867L4.41336 12.2043C4.48979 12.2867 4.53092 12.3958 4.52795 12.5083V13.8056Z"
                  fill="#FB118E"
                />
                <path
                  d="M9.99737 12.4943C9.86205 12.7005 9.6623 12.8164 9.43032 12.8164C9.19191 12.8164 9.00504 12.7198 8.83106 12.4943L7.31036 10.6385C7.20082 10.5032 7.14282 10.3614 7.14282 10.2068C7.14282 9.88458 7.38768 9.63327 7.70342 9.63327C7.89673 9.63327 8.05138 9.70415 8.20603 9.90391L9.40455 11.4311L11.9498 7.34577C12.0851 7.12669 12.2591 7.02359 12.4524 7.02359C12.7553 7.02359 13.0388 7.23623 13.0388 7.55197C13.0388 7.70017 12.9615 7.85482 12.8777 7.99014L9.99737 12.4943Z"
                  fill="white"
                />
              </svg>
            </div>
            <div
              class="c11 css-1tiu9da"
            >
              Azuki #3318
            </div>
          </div>
          <div
            class="c12 c13"
            width="min-content"
          >
            <button
              class="c14 c15 c16 c17"
            >
              Make an offer
            </button>
          </div>
        </div>
      </div>
      <div
        class="c4 c5 c18"
      >
        <div
          class="c1 c19"
        >
          <div
            class="c20"
          >
            <div
              class="c4 c5 c21"
            >
              <div
                class="c11 c22 css-rjqmed"
              >
                <div
                  class="c4 c23"
                >
                  Description
                </div>
              </div>
              <div
                class="c11 c24 css-rjqmed"
              >
                <div
                  class="c4 c23"
                >
                  Details
                </div>
              </div>
            </div>
            <div
              class="c25"
            >
              Description Content
            </div>
          </div>
        </div>
        <div
          class="c20"
        >
          <div
            class="c4 c5 c21"
          >
            <div
              class="c11 c22 css-rjqmed"
            >
              <div
                class="c4 c23"
              >
                Activity
              </div>
            </div>
            <div
              class="c11 c24 css-rjqmed"
            >
              <div
                class="c4 c23"
              >
                Offers
                <div
                  class="c26 css-f8aq60"
                >
                  10+
                </div>
              </div>
            </div>
            <div
              class="c11 c24 css-rjqmed"
            >
              <div
                class="c4 c23"
              >
                Listings
              </div>
            </div>
          </div>
          <div
            class="reset_base__1klryar0 sprinkles_display_flex_sm__rgw6ez44v sprinkles_flexDirection_row_sm__rgw6ez471 sprinkles_alignItems_center_sm__rgw6ez3j c27"
          >
            <span>
              Activity Content
            </span>
            <div
              class="c28"
            >
              <div
                class="c4 c5 c29"
                data-testid="activity-time-period-switcher"
              >
                <div
                  class="c11 css-1adn0z6"
                >
                  1 week
                </div>
                <svg
                  class="c30"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polyline
                    points="6 9 12 15 18 9"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`data page loads without header showing 1`] = `
<DocumentFragment>
  .c4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.c12 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  justify-self: flex-end;
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
}

.c14 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: primary;
  border: 0;
  border-radius: 4px;
}

.c5 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c9 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  gap: 4px;
}

.c13 {
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  gap: 12px;
}

.c23 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  gap: 8px;
}

.c10 {
  color: #7780A0;
}

.c11 {
  color: #0D111C;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c20 {
  background: #FFFFFF;
  border: 1px solid #D2D9EE;
  border-radius: 16px;
  padding: 16px 20px;
  width: 100%;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
}

.c21 {
  gap: 32px;
  width: 100;
  margin-bottom: 12px;
  padding-bottom: 12px;
  border-bottom: 1px solid #D2D9EE;
}

.c22 {
  color: #0D111C;
  line-height: 24px;
  cursor: pointer;
}

.c22:hover {
  opacity: 0.6;
}

.c24 {
  color: #98A1C0;
  line-height: 24px;
  cursor: pointer;
}

.c24:hover {
  opacity: 0.6;
}

.c26 {
  background: #D2D9EE;
  border-radius: 4px;
  padding: 2px 4px;
  color: #7780A0;
  line-height: 12px;
}

.c25 {
  height: 252px;
}

.c15 {
  padding: 16px;
  width: 100%;
  font-weight: 500;
  text-align: center;
  border-radius: 20px;
  outline: none;
  border: 1px solid transparent;
  color: #0D111C;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  position: relative;
  z-index: 1;
  will-change: transform;
  -webkit-transition: -webkit-transform 450ms ease;
  -webkit-transition: transform 450ms ease;
  transition: transform 450ms ease;
  -webkit-transform: perspective(1px) translateZ(0);
  -ms-transform: perspective(1px) translateZ(0);
  transform: perspective(1px) translateZ(0);
}

.c15:disabled {
  opacity: 50%;
  cursor: auto;
  pointer-events: none;
}

.c15 > * {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c15 > a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c16 {
  background-color: #F5F6FC;
  color: #7780A0;
  font-size: 16px;
  font-weight: 500;
}

.c16:hover {
  background-color: #d2daf7;
}

.c16:active {
  background-color: #bdc8f3;
}

.c17 {
  white-space: nowrap;
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-radius: 16px;
}

.c6 {
  gap: 24px;
}

.c7 {
  width: 96px;
  height: 96px;
  border-radius: 20px;
  object-fit: cover;
}

.c8 {
  gap: 4px;
  margin-right: auto;
}

.c28 {
  position: relative;
}

.c29 {
  gap: 4px;
  padding: 8px;
  cursor: pointer;
  border-radius: 12px;
  width: 92px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background: #E8ECFB;
  -webkit-transition: opacity 250ms ease;
  transition: opacity 250ms ease;
}

.c29:hover {
  opacity: 0.6;
}

.c29:active {
  opacity: 0.4;
}

.c30 {
  height: 16px;
  width: 16px;
  color: #7780A0;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: -webkit-transform 125ms;
  -webkit-transition: transform 125ms;
  transition: transform 125ms;
}

.c27 {
  height: 568px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c0 {
  padding: 24px 64px;
  height: 100vh;
  width: 100%;
}

.c2 {
  height: 100%;
  width: 100%;
  gap: 36px;
  max-width: 1200px;
  margin: 0 auto;
}

.c3 {
  position: -webkit-sticky;
  position: sticky;
  top: 72px;
  padding-top: 16px;
  -webkit-backdrop-filter: blur(12px);
  backdrop-filter: blur(12px);
  z-index: 1;
  -webkit-transition: opacity 125ms;
  transition: opacity 125ms;
  opacity: 0;
}

.c18 {
  gap: 24px;
  padding-bottom: 45px;
}

.c19 {
  gap: 24px;
  width: 100%;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
}

@media screen and (max-width:1024px) {
  .c7 {
    display: none;
  }
}

@media screen and (max-width:768px) {
  .c0 {
    height: 100%;
  }
}

@media screen and (max-width:640px) {
  .c0 {
    padding: 24px 48px;
  }
}

@media screen and (max-width:396px) {
  .c0 {
    padding: 24px 20px;
  }
}

@media screen and (max-width:768px) {
  .c3 {
    display: none;
  }
}

@media screen and (max-width:1024px) {
  .c18 {
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
  }
}

<div
    class="c0"
  >
    <div
      class="c1 c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4 c5 c6"
        >
          <img
            class="c7"
            src="https://cdn.center.app/1/0xED5AF388653567Af2F388E6224dC7C4b3241C544/3318/50ed67ad647d0aa0cad0b830d136a677efc2fb72a44587bc35f2a5fb334a7fdf.png"
          />
          <div
            class="c1 c8"
          >
            <div
              class="c4 c9"
            >
              <div
                class="c10 css-1aekuku"
              >
                Azuki
              </div>
              <svg
                fill="none"
                height="16px"
                viewBox="0 0 20 20"
                width="16px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4.52795 13.8056C4.52719 14.4043 4.6712 14.8474 4.95997 15.135C5.24798 15.4233 5.68496 15.5651 6.27091 15.5605H7.57497C7.62945 15.5585 7.68379 15.5676 7.73463 15.5873C7.78547 15.607 7.83176 15.6369 7.87062 15.6752L8.79884 16.5928C9.22054 17.0142 9.63382 17.2237 10.0387 17.2214C10.4436 17.2191 10.8569 17.0096 11.2786 16.5928L12.1954 15.6752C12.2356 15.6365 12.2832 15.6063 12.3354 15.5866C12.3876 15.5669 12.4433 15.558 12.499 15.5605H13.7951C14.3871 15.5613 14.8283 15.4171 15.1186 15.1281C15.4089 14.839 15.5541 14.3959 15.5541 13.7987V12.5014C15.5511 12.389 15.5923 12.2799 15.6687 12.1974L16.5854 11.2798C17.0125 10.86 17.2245 10.4467 17.2214 10.0399C17.2184 9.63305 17.0064 9.21935 16.5854 8.79878L15.6687 7.88115C15.592 7.79886 15.5509 7.68965 15.5541 7.57719V6.2799C15.5533 5.68191 15.4093 5.23878 15.1221 4.95049C14.8348 4.66221 14.3925 4.51806 13.7951 4.51806H12.499C12.4433 4.52036 12.3877 4.51138 12.3355 4.49168C12.2834 4.47197 12.2357 4.44193 12.1954 4.40336L11.2786 3.48574C10.8569 3.06439 10.4436 2.85487 10.0387 2.85717C9.63382 2.85946 9.22054 3.06898 8.79884 3.48574L7.87062 4.40336C7.83164 4.44148 7.78536 4.4713 7.73454 4.49101C7.68373 4.51072 7.62943 4.51993 7.57497 4.51806H6.27091C5.67961 4.51883 5.23995 4.66182 4.95194 4.94705C4.66393 5.23228 4.51992 5.67656 4.51992 6.2799V7.58063C4.52314 7.69309 4.48197 7.80229 4.40533 7.88459L3.48859 8.80222C3.06765 9.22203 2.85718 9.63572 2.85718 10.0433C2.85718 10.4509 3.07033 10.8653 3.49662 11.2867L4.41336 12.2043C4.48979 12.2867 4.53092 12.3958 4.52795 12.5083V13.8056Z"
                  fill="#FB118E"
                />
                <path
                  d="M9.99737 12.4943C9.86205 12.7005 9.6623 12.8164 9.43032 12.8164C9.19191 12.8164 9.00504 12.7198 8.83106 12.4943L7.31036 10.6385C7.20082 10.5032 7.14282 10.3614 7.14282 10.2068C7.14282 9.88458 7.38768 9.63327 7.70342 9.63327C7.89673 9.63327 8.05138 9.70415 8.20603 9.90391L9.40455 11.4311L11.9498 7.34577C12.0851 7.12669 12.2591 7.02359 12.4524 7.02359C12.7553 7.02359 13.0388 7.23623 13.0388 7.55197C13.0388 7.70017 12.9615 7.85482 12.8777 7.99014L9.99737 12.4943Z"
                  fill="white"
                />
              </svg>
            </div>
            <div
              class="c11 css-1tiu9da"
            >
              Azuki #3318
            </div>
          </div>
          <div
            class="c12 c13"
            width="min-content"
          >
            <button
              class="c14 c15 c16 c17"
            >
              Make an offer
            </button>
          </div>
        </div>
      </div>
      <div
        class="c4 c5 c18"
      >
        <div
          class="c1 c19"
        >
          <div
            class="c20"
          >
            <div
              class="c4 c5 c21"
            >
              <div
                class="c11 c22 css-rjqmed"
              >
                <div
                  class="c4 c23"
                >
                  Description
                </div>
              </div>
              <div
                class="c11 c24 css-rjqmed"
              >
                <div
                  class="c4 c23"
                >
                  Details
                </div>
              </div>
            </div>
            <div
              class="c25"
            >
              Description Content
            </div>
          </div>
        </div>
        <div
          class="c20"
        >
          <div
            class="c4 c5 c21"
          >
            <div
              class="c11 c22 css-rjqmed"
            >
              <div
                class="c4 c23"
              >
                Activity
              </div>
            </div>
            <div
              class="c11 c24 css-rjqmed"
            >
              <div
                class="c4 c23"
              >
                Offers
                <div
                  class="c26 css-f8aq60"
                >
                  10+
                </div>
              </div>
            </div>
            <div
              class="c11 c24 css-rjqmed"
            >
              <div
                class="c4 c23"
              >
                Listings
              </div>
            </div>
          </div>
          <div
            class="reset_base__1klryar0 sprinkles_display_flex_sm__rgw6ez44v sprinkles_flexDirection_row_sm__rgw6ez471 sprinkles_alignItems_center_sm__rgw6ez3j c27"
          >
            <span>
              Activity Content
            </span>
            <div
              class="c28"
            >
              <div
                class="c4 c5 c29"
                data-testid="activity-time-period-switcher"
              >
                <div
                  class="c11 css-1adn0z6"
                >
                  1 week
                </div>
                <svg
                  class="c30"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polyline
                    points="6 9 12 15 18 9"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
